yasnippet.el

```diff
@@ -3910,40 +3910,67 @@ Meant to be called in a narrowed buffer, does various passes"
     (goto-char parse-start)
     (yas--indent snippet)))
 
+(defun yas--regexp-quote-flexspace (beg end)
+  "Return regexp match text from BEG to END allowing for whitespace changes."
+  (let ((regexps nil)
+        (pt beg))
+    (save-excursion
+      (goto-char pt)
+      (while (re-search-forward "[[:space:]]+" end t)
+        (let ((spaces (match-string-no-properties 0)))
+          (push (regexp-quote (buffer-substring-no-properties
+                               pt (match-beginning 0)))
+                regexps)
+          (setq pt (point))
+          ;; Prefer exact match, but allow some fuzzyness for spaces.
+          (push (concat "\\(?:" spaces "\\|[[:space:]]*\\)")
+                regexps)))
+      (push (regexp-quote (buffer-substring-no-properties
+                           pt end))
+            regexps))
+    (apply #'concat (nreverse regexps))))
+
 (defun yas--indent-region (from to snippet)
   "Indent the lines between FROM and TO with `indent-according-to-mode'.
 The SNIPPET's markers are preserved."
-  ;;; Apropos indenting problems....
-  ;;
-  ;; `indent-according-to-mode' uses whatever `indent-line-function'
-  ;; is available. Some implementations of these functions delete text
-  ;; before they insert. If there happens to be a marker just after
-  ;; the text being deleted, the insertion actually happens after the
-  ;; marker, which misplaces it.
-  ;;
-  ;; This would also happen if we had used overlays with the
-  ;; `front-advance' property set to nil.
-  ;;
-  ;; This is why I have these `trouble-markers', they are the ones at
-  ;; the first non-whitespace char at the line.  After indentation
-  ;; takes place we should be at the correct to restore them.  All
-  ;; other non-trouble-markers should have been *pushed* and don't
-  ;; need special attention.
   (let* ((snippet-markers (yas--collect-snippet-markers snippet))
          (to (set-marker (make-marker) to)))
     (save-excursion
       (goto-char from)
       (save-restriction
         (widen)
-        ;; Indent each non-empty line.
+        ;; Indent each non-empty line.  `indent-according-to-mode'
+        ;; uses whatever `indent-line-function' is available.  Some
+        ;; implementations of these functions delete text before they
+        ;; insert, some make complicated regexp replacements (looking
+        ;; at you, org-mode).  To find place where the marker "should"
```

Comment:
Yeah, it's a bit ridiculous, but I think this is the best way, given the variety of indentation implementations. Though the code can be simplified a bit by using `split-string` instead of the explicit loop I have right now.